janus_core/helpers/correlator.py

```diff
@@ -209,6 +212,8 @@ def __init__(
 
         Parameters
         ----------
+        n_atoms : int
+            Number of possible atoms to track.
         a : tuple[Observable, tuple, dict]
             Getter for a and kwargs.
```

Comment:
This isn't new but this (and `b`) needs updating to `Union[Observable, tuple[Observable, tuple, dict]]`.

The docstring probably could be clarified more too.

It might just be me, but I read "getter" as a function that might return the class, whereas this seems to be expecting the observable itself to be given?

It's probably also worth clarifying the tuple and dict optional components e.g. whether only one is allowed (it looks like it from the tests `"a": Velocity(["x", "y", "z"], na),`) and what each actually corresponds to.

Reply:
I will clarify those points.

On the "getter" though this exists solely to keep (after restructuring I have done here) the ability for a user to pass an arbitrary (argument conforming) function to create a custom Observable without defining a class. Maybe it is just clutter but a better name can be found. My thinking is it is a "getter of the observed value". Any suggestions welcome.

Reply:
On ```Union[Observable, tuple[Observable, tuple, dict]]```. The idea was to support an bare Observable,

```Python
"a": Velocity(["x", "y", "z"], na)
```

but also an Observable that may have additional args and kwargs

```Python
"a": (SomeObservable(["x", "y", "z"], na), (arg1, arg2, ...), {"some kwarg": 0.1, ...}
```

Thinking about it though these are more complicated and not more useful than designing an Observable class that stores some args and kwargs on initialisation and applying them in ```__call__```, as they would need to be anyway. So I think they can also be removed. We could then have ```__call__(self, atoms: Atoms) -> list[float]``` too.

The only way I can see this being useful is if MD kept track of some args and kwargs for each Observable (that vary over a simulation) and passed them into ```Correlation.update(self, atoms: Atoms)``` but I cannot think of a use case.

It came from having the ability for a user to write a function (as apposed to a class deriving from ```Observable```) and pass that as an ```Observable``` where args and kwargs are useful.